### DISCUSSION TITLE
How to get power usage from my boiler?

### DISCUSSION THREAD

**Discussion by kloknibor** [john30/ebusd#1126] [Q&A]
Hi All,

I have an Valliant ecoTEC plus VHR 25-32CS/1-5 CW4 Boiler. I was pleasantly surprised to find out that on the screen it shows the power and gas usage. So for this reason I was hoping to read this out remotely and keep track of it.

However I have tried all ebusctl commands and this was the response I got for all power related registers. As you can see it seems like the boiler respons with an empty request for al the power related statistics:

> bai PrEnergyCountHc1 =  (ERR: invalid position for 3108b509030df600 / 00)
> bai PrEnergyCountHc2 =  (ERR: invalid position for 3108b509030df800 / 00)
> bai PrEnergyCountHc3 =  (ERR: invalid position for 3108b509030dfa00 / 00)
> bai PrEnergyCountHwc1 =  (ERR: invalid position for 3108b509030dc600 / 00)
> bai PrEnergyCountHwc2 =  (ERR: invalid position for 3108b509030dc800 / 00)
> bai PrEnergyCountHwc3 =  (ERR: invalid position for 3108b509030dca00 / 00)
> bai PrEnergySumHc1 =  (ERR: invalid position for 3108b509030df500 / 00)
> bai PrEnergySumHc2 =  (ERR: invalid position for 3108b509030df700 / 00)
> bai PrEnergySumHc3 =  (ERR: invalid position for 3108b509030df900 / 00)
> bai PrEnergySumHwc1 =  (ERR: invalid position for 3108b509030dc500 / 00)
> bai PrEnergySumHwc2 =  (ERR: invalid position for 3108b509030dc700 / 00)
> bai PrEnergySumHwc3 =  (ERR: invalid position for 3108b509030dc900 / 00)

here is the output of ebusctl info

> root@ebusd:~# ebusctl info            
> version: ebusd 23.3.23.3
> update check: OK
> device: 192.168.1.40:9999, TCP, enhanced, firmware 1.1[4106].1[4106]
> signal: acquired
> symbol rate: 41
> max symbol rate: 108
> min arbitration micros: 0
> max arbitration micros: 16
> min symbol latency: 8
> max symbol latency: 23
> scan: finished
> reconnects: 0
> masters: 3
> messages: 225
> conditional: 3
> poll: 0
> update: 10
> address 03: master #11
> address 08: slave #11, scanned "MF=Vaillant;ID=BAI00;SW=1103;HW=7603", loaded "vaillant/bai.308523.inc", "vaillant/08.bai.csv"
> address 31: master #8, ebusd
> address 36: slave #8, ebusd
> address 7f: master #24
> address 84: slave #24, scanned "MF=Vaillant;ID=OTS00;SW=0205;HW=9101"

Is there anything what I can do to see if I can retrieve the power usage somehow? Or do I just need to accept that I won't be able to retrieve the power consumption over e-bus at all?

Thanks all!

**Comment by G8GYW** [comment]
Not sure how you're trying to read the data but this works for me on my ecoTEC Plus 418:

`pi@boiler:~ $ ebusctl read PrEnergyCountHc1
2389126

pi@boiler:~ $ ebusctl read PrEnergyCountHwc1
481788
`
Mike
